### ISSUE TITLE
SEP-24 & 6: Add `pending_user_transfer_complete` transaction status

### ISSUE THREAD

**Issue by JakeUrban** [stellar/stellar-protocol#1083]
### What problem does your feature solve?

Neither SEP-6 or SEP-24 have a status that indicates the user must take action in order to complete the transaction, the most common use case being a user who needs to pick up cash from a retail location.

`pending_user` exists but may be too broad, since it is a catch all for any pending-user-action state, and could conceivably  be used multiple times in a transaction's lifecycle for different purposes.

### What would you like to see?

`pending_user_transfer_complete` added to the set of standardized statuses.

- `pending_user_transfer_start`: the user needs to provide funds to the anchor to initiate the transaction
- `pending_user_transfer_complete`: the user needs to collect funds from the anchor to complete the transaction

### What alternatives are there?

Using `pending_user`.

cc @leighmcculloch

**Comment by leighmcculloch** [comment]
What's the next state after the new status? Does complete mean that it is ready for something? Maybe we need a new status that indicates what it is ready for?

**Comment by JakeUrban** [comment]
The idea is that it would go from `pending_user_transfer_complete` to `completed`, where `completed` means settlement has occurred and there is no action needed from either party.

**Comment by leighmcculloch** [comment]
I'm skeptical of `pending_user_transfer_complete` because it is speaking about what has happened, but not what the current state of things is. This makes it not particularly clear about what comes next.

It sounds like we are missing some statuses to indicate whatcomes next, such as the settlement being ready or in process. 

For example, for the cash scenario described I'm imagining:
- `pending_user` when waiting on the user to do something
- `delivery_ready` when cash can be picked up
- `completed` when cash is picked up

For example, for bank transfers:
- `pending_user` when waiting on the user to do something
- `delivery_in_progress` when settlement is queued, or has begun, or is the next thing to happen
- `completed` when settlement is complete

Possibly we could get 

Would this fit or am I not getting the context?

**Comment by JakeUrban** [comment]
`pending_user_transfer_complete` is supposed to mean that the cash can be picked up. The transfer is pending the user complete it, in the same way that `pending_user_transfer_start` is pending the user start the transfer.

The idea is for a withdrawal transaction, the status go:
- `incomplete` - the anchor still needs more info before the user can send funds, like KYC info
- `pending_user_transfer_start` - the user needs to send funds to the anchor. In the withdrawal case this is a payment to the anchor's distribution account
- `pending_anchor` - the anchor received the payment and is processing it before the user can pick up cash
- `pending_user_transfer_complete` - the cash is ready and the user has to go pick it up
- `completed` - the user picked up the case and the transaction is settled.

**Comment by leighmcculloch** [comment]
This flow is a little counterintuitive, because `pending_anchor` in name doesn't appear to have anything to do with `user_transfer`, it looks as if the user_transfer portion is already complete, so to have it then complete after pending_anchor is confusing.

I would at least change the description of the `pending_user_transact_complete` a little to focus on the user transfer since it is named for the user transaction:

```diff
-pending_user_transfer_complete - the cash is ready and the user has to go pick it up
+pending_user_transfer_complete - the user has sent the funds to the anchor and the anchor has received them
```

This is all very minor, and I defer to you. Just sharing how it occurs for me because I was mentioned 😄.

**Comment by JakeUrban** [comment]
@lijamie98 this PR was merged right? I'll close this issue and you can reopen if there is still outstanding TODOs.

**Comment by lijamie98** [comment]
Yes, it is merged. Am I supposed to close it or is it for the issue
reporter to close?

**Comment by JakeUrban** [comment]
The PR should automatically close related issues when its merged. You can use syntax like

resolves #1083

to link issues from the PR description.

**Comment by lijamie98** [comment]
Thanks for the helpful tip.
